integration-tests/README.md

```diff
@@ -0,0 +1,121 @@
+# Native image integration tests
+Builds Native image executables for small applications based on some Native image capable runtimes such
+as Quarkus, Micronaut and Helidon. It also provides a convenient way of building small test apps for Native image backbox testing.
+
+## Prerequisites
+
+The TS expects you run Java 11+ and have ```ps``` program available on your Linux/Mac
+and ```wmic``` (by default present) on your Windows system.
+Native image builds also require you have the following packages installed:
+* glibc-devel
+* zlib-devel
+* gcc
+* libffi-devel
+
+On Fedora/CentOS/RHEL they can be installed with:
+```bash
+dnf install glibc-devel zlib-devel gcc libffi-devel libstdc++-static
+```
+
+On Ubuntu-like systems with:
+```bash
+apt install gcc zlib1g-dev libffi-dev build-essential
+```
+
+## Usage
+
+The basic usage that runs all tests:
+```
+mvn clean verify -Ptestsuite
```

Comment:
Out of curiosity, why not just `mvn clean verify`?

Reply:
@zakkak The test apps are modules and they are included in the parent by default, so as you have them imported in your IDE as a one project without any hassle. When you run the test suite, you don't want to build those apps too; you just want to run the TS. The TS builds the apps at its own discretion at the right time.  Hence -Ptestsuite says "Just run the TS, don't build apps/* modules..."

If your Maven-fu could bring a better way of doing this without hampering the IDE experience, I am all ears...

Reply:
Thanks for the clarification @Karm .

If I understand correctly both `mvn clean verify -Ptestsuite` and `mvn clean verify` will have the same **result**, but `mvn clean verify -Ptestsuite` might be faster in some cases.

The difference is that `mvn clean verify` will decent in the apps and do `mvn clean verify` there as well, while `mvn clean verify -Ptestsuite` will only run in testsuite (it is actually the same with `mvn clean verify -pl testsuite`) and the test suite will implicitly invoke `mvn ... package ...` in the apps to build them as needed.

My Maven-fu ( :) ) approach would probably use `io.quarkus:quarkus-maven-plugin` for creating the native images in the apps pom files, and avoid invoking mvn from the testsuite harness.

The entry would look like this (copied from a quarkus integration test):
```xml
                    <plugin>
                        <groupId>io.quarkus</groupId>
                        <artifactId>quarkus-maven-plugin</artifactId>
                        <executions>
                            <execution>
                                <id>native-image</id>
                                <goals>
                                    <goal>native-image</goal>
                                </goals>
                                <configuration>
                                    <cleanupServer>true</cleanupServer>
                                    <enableHttpUrlHandler>true</enableHttpUrlHandler>
                                    <graalvmHome>${graalvmHome}</graalvmHome>
                                </configuration>
                            </execution>
                        </executions>
                    </plugin>
```

Not sure it's worth the extra effort though. Just sharing my thoughts.

Reply:
@zakkak I don't understand how this helps Helidon or Micronaut  (and eventually other modules)...? 
I might be missing the point of what is bothering you in the first place. Is it that the default profile is suited for IDE and not for execution? Why would it be a problem?

Reply:
No problem here, just discussing and sharing my thoughts, feel free to ignore me :) Sorry for not making this clear.

With this plugin you should be able to create the native-image directly through maven (for any framework), without the need to invoke `mvn` and `native-image` from the test harness. In theory this could make configuration easier as well (e.g. for using a different `native-image` executable.